feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchScreen.kt

```diff
@@ -16,80 +16,252 @@
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.withStyle
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
 import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.google.samples.apps.nowinandroid.core.designsystem.icon.NiaIcons
 import com.google.samples.apps.nowinandroid.core.designsystem.theme.NiaTheme
+import com.google.samples.apps.nowinandroid.core.domain.model.FollowableTopic
+import com.google.samples.apps.nowinandroid.core.domain.model.UserNewsResource
 import com.google.samples.apps.nowinandroid.core.ui.DevicePreviews
+import com.google.samples.apps.nowinandroid.core.ui.NewsFeedUiState
 import com.google.samples.apps.nowinandroid.core.ui.R.string
 import com.google.samples.apps.nowinandroid.core.ui.TrackScreenViewEvent
+import com.google.samples.apps.nowinandroid.core.ui.TrackScrollJank
+import com.google.samples.apps.nowinandroid.core.ui.newsFeed
+import com.google.samples.apps.nowinandroid.feature.foryou.ForYouViewModel
+import com.google.samples.apps.nowinandroid.feature.interests.InterestsViewModel
+import com.google.samples.apps.nowinandroid.feature.interests.TopicsTabContent
 import com.google.samples.apps.nowinandroid.feature.search.R as searchR
 
 @Composable
 internal fun SearchRoute(
     modifier: Modifier = Modifier,
     onBackClick: () -> Unit,
-    viewModel: SearchViewModel = hiltViewModel(),
+    onInterestsClick: () -> Unit,
+    onTopicClick: (String) -> Unit,
+    interestsViewModel: InterestsViewModel = hiltViewModel(),
+    searchViewModel: SearchViewModel = hiltViewModel(),
+    forYouViewModel: ForYouViewModel = hiltViewModel(),
 ) {
+    val uiState by searchViewModel.searchResultUiState.collectAsStateWithLifecycle()
     SearchScreen(
         modifier = modifier,
         onBackClick = onBackClick,
-        onSearchQueryChanged = viewModel::onSearchQueryChanged,
+        onFollowButtonClick = interestsViewModel::followTopic,
+        onInterestsClick = onInterestsClick,
+        onSearchQueryChanged = searchViewModel::onSearchQueryChanged,
+        onTopicClick = onTopicClick,
+        onNewsResourcesCheckedChanged = forYouViewModel::updateNewsResourceSaved,
+        uiState = uiState,
     )
 }
 
 @Composable
 internal fun SearchScreen(
     modifier: Modifier = Modifier,
     onBackClick: () -> Unit = {},
+    onFollowButtonClick: (String, Boolean) -> Unit = { _, _ -> },
+    onInterestsClick: () -> Unit = {},
+    onNewsResourcesCheckedChanged: (String, Boolean) -> Unit = { _, _ -> },
     onSearchQueryChanged: (String) -> Unit = {},
+    onTopicClick: (String) -> Unit = {},
+    uiState: SearchResultUiState = SearchResultUiState.Loading,
 ) {
+    val searchQuery = remember { mutableStateOf("") }
```

Comment:
Use directly `SearchViewModel.searchQuery`, avoid duplication

```suggestion
    val searchQuery = searchViewModel.searchQuery.collectAsStateWithLifecycle()
```

Reply:
Done